gui.py

```diff
@@ -113,6 +116,18 @@ def register_speed_provider(self, provider: SpeedProvider):
     def register_fuel_provider(self, provider: FuelProvider):
         self.fuel_display.repeat(1000, self.__updateFuel, args=[provider])
 
+    ##obd/gps widget - no functionality yet, but the alternate images are ready to go
+    def create_gps_obd_icon_widget(self, parent):
+        result = Box(parent, width=208, height=48)
+        result.set_border(1, "darkgreen")
+        Picture(result, image="resources/images/gps_ok.gif", align="left")
+        Picture(result, image="resources/images/obd_off.gif", align="right")
+
+        ##hidden icon alternatives
+        Picture(result, image="resources/images/gps_off.gif", align="left", visible=False)
+        Picture(result, image="resources/images/obd_ok.gif", align="right", visible=False)
```

Comment:
It's not ideal that we only are able to reference these child objects by their array offsets.
It makes for clumsy/buggy code when we have to write:
`self.gps_obd_icon_widget.children[3].visible = true`

What would be nicer would be to create a subclassed Picture, and add appropriate methods onto the object itself, for example:

```python
class EnableDisablePicture(Picture):
    def __init__(self, parent, on_image, off_image):
       Picture.__init__(parent, image=off_image, align="left")
       self.on_image = on_image
       self.off_image = off_image

   def enable(self):
      self.image = self.on_image

  def disable(self):
      self.image = self.off_image
```

Then we can create the two buttons in the main gui creation code with
```python
    self.obd_image = EnableDisablePicture(col2, "resources/images/obd_ok.gif", "resources/images/obd_off.gif")
```
It would make for nicer enable/disable callback logic that way.

Obviously this problem is widespread, but it would be nice to clean it up in the future.